### PR TITLE
TM anomaly: new modes: likelihood,...

### PR DESCRIPTION
**Pull request by breznak**
For #469 , and https://github.com/htm-community/NAB/pull/5 

- [x] implement other modes for `TM.anomaly`
  - `ANMode {DISABLED, RAW, LIKELIHOOD, LOGLIKELIHOOD}`
  - this is using the not 100% validated c++ Likelihood, but provides convenient API to use it within TM,
    and aims to replace code in NAB HTMcore detector
- [x] bidings
- [x] still some bug which segfaults

### CODE REVIEW

**Comment by breznak** [comment]
Ping..

File: src/htm/algorithms/TemporalMemory.hpp [hunk 1]
```diff
@@ -625,21 +645,34 @@ class TemporalMemory : public Serializable
...
   /*
    *  anomaly score computed for the current inputs
    *  (auto-updates after each call to TM::compute())
    *
    *  @return a float value from computeRawAnomalyScore()
    *  from Anomaly.hpp
    */
-  const Real &anomaly = anomaly_;
+const Real &anomaly = tmAnomaly_.anomaly_; //this is position dependant, the struct anomaly_tm must be defined before this use,
+// otherwise this surprisingly compiles, but a call to `tmAnomaly_.anomaly` segfaults!
```
Comment (ctrl-z-9000-times) [inline src/htm/algorithms/TemporalMemory.hpp]: wow, Interesting ...
This seems like a problem with the C++ language, or maybe the compiler?  Thanks for making note of this issue.

Comment (breznak) [reply src/htm/algorithms/TemporalMemory.hpp]: yep, strange thing, kept me busy for a while. The compiler might be, I've been using g++-9 recently due to the ParallelTS.

**Comment by breznak** [comment]
Thank you for reviews, I'll now be able to use this code in our NAB detector.

**Pull request merged**